r/src/arrow_metadata.h

```diff
@@ -0,0 +1,38 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "./arrow_types.h"
+
+#if defined(ARROW_R_WITH_ARROW)
+#include <arrow/util/key_value_metadata.h>
+
+inline std::shared_ptr<arrow::KeyValueMetadata> KeyValueMetadata__Make(
```

Comment:
Two thoughts:

1. We already have KeyValueMetadata support in R, so if these helpers are necessary, I'd expect to see them used in those other places too.
2. It looks like these functions are opposites, one takes a named character vector and turns it into arrow::KVM, and the other takes KVM and makes a named character vector. Can we make the function names more obvious about that?

Reply:
1. I can replace other instances of KeyValueMetadata wrangling to use these helpers, or I could just leave things inlined as in other files under `arrow/r/`
2. How about `KeyValueMetadata__from_strings`, `KeyValueMetadata__to_strings`?

Reply:
Yes, if these helpers are useful, let's use them everywhere. And yeah I like those names better, thanks.